acoustid/data/source.py

```
@@ -22,32 +23,63 @@ def get_source(conn: AppDB, id: int) -> dict[str, Any] | None:
+                "Inserted source %d with account %d and application %d (%s)",
+                id,
+                account_id,
+                application_id,
+                version,
+            )
+            return source_id
+
+        # If insert returned NULL, it means there was a conflict, so we need to query again
+        retries -= 1
+        if retries == 0:
+            raise RuntimeError("Failed to insert source")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)